client-projects/the-lumery/scss/button.scss

```diff
@@ -0,0 +1,186 @@
+// .btn is in fact (in the tradition of Webflow) an <a> tag rather than a <button>
+// there's a pre-existing class of .button, hence the name .btn
+// these components have the following structure:
+// <a class="btn">
+//   <div class="btn__bg">
+//   <div class="btn__text">   (Webflow won't allow a <p> or a <span>)
+//</a>
+
+.btn {
+  --easing: cubic-bezier(0.1, 0, 0.3, 1);
+  --border-radius: 100px;
+  position: relative;
+  box-sizing: border-box;
+  border-radius: var(--border-radius);
+  display: grid;
+  grid-template-areas: "button";
+
+  cursor: pointer;
+  text-decoration: none;
+
+  // visible focus will be applied on the .btn__bg element instead
+  &:focus {
+    outline: 0;
+  }
+}
+
+.btn__bg {
+  grid-area: button;
+  width: 100%;
+  height: 100%;
+  border-radius: var(--border-radius);
+  transform-origin: center;
+  transform: scale(1);
+  transition: transform 300ms var(--easing);
+  outline-color: transparent;
+  outline-width: 2px;
+  outline-style: solid;
+
+  .btn:hover &,
+  .btn:active &,
+  .card__link:hover &,
+  .card__link:active &,
+  .button--submit:hover + .fake-button-container &,
+  .button--submit:active + .fake-button-container & {
+    transform: scale(1.1);
+  }
+}
+
+// in a primary button, a yellow background div is covered with its black ::after
+// pseudoelement on hover.
+.btn__bg--primary {
+  background-color: var(--color-yellow);
+
+  // black bg
+  &::after {
+    content: "";
+    grid-area: button;
+    display: block;
+    border-radius: var(--border-radius);
+    width: 100%;
+    height: 100%;
+    background-color: var(--color-black);
+    transform-origin: center;
+    transform: scaleX(0);
+    transition: transform 300ms var(--easing);
+  }
+
+  // black background expands out horizontally from centre
+  .btn:hover &::after,
+  .card__link:hover &::after,
+  .button--submit:hover + .fake-button-container &::after {
+    transform: scaleX(1);
+  }
+
+  .btn:focus &,
+  .card__link:focus &,
+  .button--submit:focus + .fake-button-container & {
+    outline-color: var(--color-black);
+  }
+
+  // the hovered button would disappear on a black background, so:
+  .background-color-black .btn:hover & {
+    border: 1px solid var(--color-yellow);
+  }
+  .background-color-black .btn:focus & {
+    outline: 2px solid var(--color-white);
+  }
+}
+
+// in a secondary button, there's no background, just a contrasting border
+.btn__bg--secondary {
+  border-width: 1px;
+  border-style: solid;
+  border-color: var(--color-black);
```

Comment:
Any reason not to go with the `border` shorthand here like you did above?

Reply:
I've got a reason, but maybe it only makes sense to me!
- Where I've used the shorthand (for `.background-color-black .btn:hover &`) there is no `border` on the button ordinarily, so I'm setting all of these properties at once.
- in the case of `.btn__bg--secondary`, I'm going to be changing the `border-color` value based on `focus` and `hover`- the `width` and `style` aren't going to change so I only want to list what _is_ changing.

Reply:
I get that you only want to change the border's colour below, but doing this bit as a 1 liner (and leaving the subsequent border-color overrides as-is) doesn't stop that working. But whatever, just an observation. Totally valid either way :)

Reply:
they are applied to different elements though — the one-liner is on the primary button, the longhand is on the secondary button. I'm not doing it differently for functionality, they both work the same, but it's about making the code clear. 
Maybe it's only clear to me, I don't know!